src/server/routes/parcel.routes.spec.ts

```diff
@@ -0,0 +1,112 @@
+import type { FastifyInstance, InjectOptions } from 'fastify';
+import {
+  generateIdentityKeyPairSet,
+  generatePDACertificationPath,
+} from '@relaycorp/relaynet-testing';
+import { subDays } from 'date-fns';
+
+import { configureMockEnvVars, REQUIRED_ENV_VARS } from '../../testUtils/envVars.js';
+import { makeTestPohttpServer } from '../../testUtils/pohttpServer.js';
+import { type MockLogSet, partialPinoLog } from '../../testUtils/logging.js';
+import type { InternetEndpoint } from '../../utilities/awala/InternetEndpoint.js';
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+import { generateParcel } from '../../testUtils/awala/parcel.js';
+
+configureMockEnvVars(REQUIRED_ENV_VARS);
+
+describe('parcel route', () => {
+  const getTestServerFixture = makeTestPohttpServer();
+  let server: FastifyInstance;
+  let logs: MockLogSet;
+  let activeEndpoint: InternetEndpoint;
+
+  const validRequestOptions: InjectOptions = {
+    headers: {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      'Content-Type': 'application/vnd.awala.parcel',
+    },
+
+    method: 'POST',
+    payload: {},
+    url: '/',
+  };
+  beforeEach(async () => {
+    ({ server, logs } = getTestServerFixture());
+    activeEndpoint = await server.getActiveEndpoint();
+  });
+
+  test('Valid parcel should be accepted', async () => {
+    const parcelRecipient = {
+      id: activeEndpoint.id,
+      internetAddress: activeEndpoint.internetAddress,
+    };
+    const keyPairSet = await generateIdentityKeyPairSet();
+    const certificatePath = await generatePDACertificationPath(keyPairSet);
+    const { parcelSerialized } = await generateParcel(
+      parcelRecipient,
+      certificatePath.privateEndpoint,
+      keyPairSet,
+    );
+
+    const response = await server.inject({
+      ...validRequestOptions,
+      payload: parcelSerialized,
+    });
+
+    expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.ACCEPTED);
+  });
+
+  test('Content-Type other than application/vnd.awala.parcel should be refused', async () => {
+    const response = await server.inject({
+      ...validRequestOptions,
+
+      headers: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        'Content-Type': 'application/json',
+      },
+    });
+
+    expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.UNSUPPORTED_MEDIA_TYPE);
```

Comment:
Missing log check.

Reply:
It's checked by the contentTypeParser and is automatically declined by fastify. Shall we try and inject logger?

Reply:
I see. In that case, let's ignore my comment.